File: 프로그래머스/2/131127. 할인 행사/할인 행사.swift
import Foundation

func solution(_ want:[String], _ number:[Int], _ discount:[String]) -> Int {
    var dict: [String: Int] = [:]
    for (item, num) in zip(want, number) {
        dict[item, default: 0] = num
    }
    var ret = 0
    for start in 0...discount.count - 10 {
        var window: [String: Int] = [:]
        for i in start..<(start + 10) {
            window[discount[i], default: 0] += 1
        }
        var flag = true
        for (value, count) in dict {
            if window[value, default: 0] < count {
                flag = false
                break
            }
        }
        if flag { ret += 1 }
    } 
    return ret
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int n = (int)want.size();
    int answer = 0;
    map<string, int> mp;
    for(int i = 0; i < n; i++) {
        mp[want[i]] += number[i];
    }
    
    for(int i = 0; i <= discount.size() - 10; i++) {
        map<string, int> mp2;
        for(int j = i; j < i+10; j++) {
            mp2[discount[j]]++;
        }
        if(mp == mp2) answer++;
    }
    
    return answer;
}